lib/reek/smells/smell_detector.rb

```diff
@@ -60,6 +63,12 @@ def enabled_for?(context)
         config.enabled? && config_for(context)[SmellConfiguration::ENABLED_KEY] != false
       end
 
+      def check_suppression(context)
+        return [] unless sniff(context).empty?
```

Comment:
I think an "unless" with another negative like "empty?" is hard to read.
How about turning this into a positive like

```Ruby
return [] if sniff(context) > 0
```

Or even better: You are doing the `sniff(context)` in **any** case.
How about doing doing `sniff(context)` once in `run_for` above and then use it here?